### PR TITLE
User/asvpxzont/authentication

### PR DESCRIPTION
**Pull request by asvpxzont**

### CODE REVIEW

File: server/models/user.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+});
+
+// create all the defined tables in the specified database.
+sequelize.sync()
+    .then(() => console.log('users table has been successfully created, if one doesn\'t exist'))
+    .catch(error => console.log('This error occured', error));
+
+// export User model for use in other files.
+module.exports = User;
```
Comment (rbUUbr) [inline server/models/user.js]: Add empty line at the end

File: server/models/user.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            return bcrypt.compareSync(password, this.password);
+        }
+    }
+});
+
+// create all the defined tables in the specified database.
+sequelize.sync()
+    .then(() => console.log('users table has been successfully created, if one doesn\'t exist'))
+    .catch(error => console.log('This error occured', error));
+
```
Comment (rbUUbr) [inline server/models/user.js]: make intendation `2`, not `4`

File: server/config/config.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "development": {
-    "username": "postgres",
+    "username": "asvpxzont",
```
Comment (rbUUbr) [inline server/config/config.json]: read about `dotenv` library and move username and password to `.env vars`

File: server.js [hunk 4]
```diff
@@ -7,4 +12,118 @@ app.get('/api/hello', (req, res) => {
...
+    }
+});
+
+
+// route for handling 404 requests(unavailable routes)
+app.use(function (req, res, next) {
+    res.status(404).send("Sorry can't find that!")
+});
+
 app.listen(port, () => console.log(`Listening on port ${port}`));
```
Comment (rbUUbr) [inline server.js]: add empty line at the end

File: server.js [hunk 5]
```diff
@@ -7,4 +12,118 @@ app.get('/api/hello', (req, res) => {
...
+app.get('/logout', (req, res) => {
+    if (req.session.user && req.cookies.user_sid) {
+        res.clearCookie('user_sid');
+        res.redirect('/');
+    } else {
+        res.redirect('/login');
+    }
+});
+
+
```
Comment (rbUUbr) [inline server.js]: remove one empty line

File: server.js [hunk 6]
```diff
@@ -7,4 +12,118 @@ app.get('/api/hello', (req, res) => {
...
+// route for user's dashboard
+app.get('/dashboard', (req, res) => {
+    if (req.session.user && req.cookies.user_sid) {
+        res.sendFile(__dirname + '/client/public/dashboard.html');
+    } else {
+        res.redirect('/login');
+    }
+});
+
+
```
Comment (rbUUbr) [inline server.js]: remove one empty line

File: server.js [hunk 7]
```diff
@@ -7,4 +12,118 @@ app.get('/api/hello', (req, res) => {
...
+                res.redirect('/dashboard');
+            })
+            .catch(error => {
+                res.redirect('/signup');
+            });
+    });
+
+
+// route for user Login
+app.route('/login')
```
Comment (rbUUbr) [inline server.js]: we are using `app.http_method` notation

File: server.js [hunk 8]
```diff
@@ -7,4 +12,118 @@ app.get('/api/hello', (req, res) => {
...
+    key: 'user_sid',
+    secret: 'somerandonstuffs',
+    resave: false,
+    saveUninitialized: false,
+    cookie: {
+        expires: 600000
+    }
+}));
+
+// This middleware will check if user's cookie is still saved in browser and user is not set, then automatically log the user out.
```
Comment (rbUUbr) [inline server.js]: remove copied comments00)0)

File: server.js [hunk 9]
```diff
@@ -7,4 +12,118 @@ app.get('/api/hello', (req, res) => {
...
+app.use(cookieParser());
+
+// initialize express-session to allow us track the logged-in user across sessions.
+app.use(session({
+    key: 'user_sid',
+    secret: 'somerandonstuffs',
+    resave: false,
+    saveUninitialized: false,
+    cookie: {
+        expires: 600000
```
Comment (rbUUbr) [inline server.js]: move to `.env` variable

File: server/models/user.js (hunk 1 above)
Comment (asvpxzont) [reply server/models/user.js]: done

File: server/models/user.js (hunk 2 above)
Comment (asvpxzont) [reply server/models/user.js]: done

File: server/config/config.json (hunk 3 above)
Comment (asvpxzont) [reply server/config/config.json]: done

File: server.js (hunk 4 above)
Comment (asvpxzont) [reply server.js]: done

File: server.js (hunk 5 above)
Comment (asvpxzont) [reply server.js]: done

File: server.js (hunk 6 above)
Comment (asvpxzont) [reply server.js]: done

File: server.js (hunk 7 above)
Comment (asvpxzont) [reply server.js]: done

File: server.js (hunk 8 above)
Comment (asvpxzont) [reply server.js]: done

File: server.js (hunk 9 above)
Comment (asvpxzont) [reply server.js]: done

File: server.js [hunk 10]
```diff
@@ -1,10 +1,40 @@
 const express = require('express');
+var bodyParser = require('body-parser');
```
Comment (rbUUbr) [inline server.js]: @asvpxzont 
change `var` to `let`

**Pull request merged**